src/makeInfinityAdder.js

```diff
@@ -6,7 +6,35 @@
  */
 
 function makeInfinityAdder() {
-  // write code here
+  let sum = 0;
+
+  const addition = (x) => {
+    if (x === undefined) {
+      const result = sum;
+
+      sum = 0;
+
+      return result;
+    } else {
+      sum += x;
+    }
+
+    return function(y) {
+      if (y === undefined) {
+        const result = sum;
+
+        sum = 0;
+
+        return result;
+      } else {
+        sum += y;
+
+        return addition;
+      }
+    };
+  };
+
+  return addition;
 }
 
```

Comment:
no needed else statement after return
```suggestion
  const addition = (x) => {
    if (x === undefined) {
      const result = sum;

      sum = 0;

      return result;
    } 
      sum += x;
    
        return addition;
    };
  };

  return addition;
}

```

do you really need the second function?

Reply:
fixed